internal/cortex/querier/stats/stats.go

```diff
@@ -89,6 +89,38 @@ func (s *Stats) LoadFetchedChunkBytes() uint64 {
 	return atomic.LoadUint64(&s.FetchedChunkBytes)
 }
 
+func (s *Stats) SetPeakSamples(peakSamples int32) {
+	if s == nil {
+		return
+	}
+
+	atomic.StoreInt32(&s.PeakLoadedSamples, peakSamples)
+}
+
+func (s *Stats) LoadPeakSamples() int32 {
+	if s == nil {
+		return 0
+	}
+
+	return atomic.LoadInt32(&s.PeakLoadedSamples)
+}
+
+func (s *Stats) AddTotalSamples(totalSamples int64) {
+	if s == nil {
+		return
+	}
+
+	atomic.AddInt64(&s.TotalLoadedSamples, totalSamples)
+}
+
+func (s *Stats) LoadTotalSamples() int64 {
+	if s == nil {
+		return 0
+	}
+
+	return atomic.LoadInt64(&s.TotalLoadedSamples)
+}
+
 // Merge the provide Stats into this one.
 func (s *Stats) Merge(other *Stats) {
```

Comment:
I am not sure why we are not using this function to add stats samples. And why this funciton is not used anywhere.
This is where Cortex tracks and merges stats https://github.com/cortexproject/cortex/blob/master/pkg/querier/stats/stats.go#L346

Reply:
Because that is Cortex "private" way of tracking statistics, but in Thanos we chose to abide to Prometheus "simpler" stats interface using only Samples Total and Peak Samples, which is implemented all the way into the thanos promql engine.

Reply:
I see so we are not calling `AddFetchedSeries`, `AddFetchedChunkBytes`, etc in Thanos querier.
This is fine. But I don't think collecting this data is conflicting of collecting the standard query stats. We can still do both.

Anyway, it is not blocking this PR so we can change if we feel useful